conda_content_trust/common.py

```diff
@@ -181,145 +179,34 @@ class MixinKey:
-    def from_bytes(cls, key_value_in_bytes):
-        """
-        Constructs an object of the class based on the given key value.
-        The "cryptography" library provides from_public_bytes() and
-        from_private_bytes() class methods for Ed25519PublicKey and
-        Ed25519PrivateKey classes in place of constructors.  We extend provide
-        a single API for those, and make the created objects objects of the
-        subclass using this mix-in.
-        """
-        # from_private_bytes() and from_public_bytes() both check length (32),
-        # but do not produce helpful errors if the argument provided it is not
-        # the right type, so we'll do that here before calling them.
-        checkformat_byteslike(key_value_in_bytes)
-
-        if issubclass(cls, ed25519.Ed25519PrivateKey):
-            new_object = cls.from_private_bytes(key_value_in_bytes)
-
-        elif issubclass(cls, ed25519.Ed25519PublicKey):
-            new_object = cls.from_public_bytes(key_value_in_bytes)
-
-        else:
-            assert False, (
-                "Code error: this should not be possible.  This mix-in "
-                "should only be used by classes inheriting from the "
-                '"cryptography" library ed25519 key classes.'
-            )
-
-        # Fixed:
-        # # TODO: ✅❌⚠️💣 Changing this here is uncouth.  It MUST BE SET AT
-        # #               CLASS DEFINITION time.  Change this!
-        # # Note that this mro modification mess is required in some form or
-        # # another because ed25519.Ed25519PrivateKey and Ed25519PublicKey
-        # # use metaclassing (in a way that I don't think is useful, btw).
-        # # This line is poking cls.__bases__.  It would appear to do nothing,
-        # # since we're extending a tuple with nothing, but it *actually* causes
-        # # the class's MRO (method resolution order) to be recalculated.
-        # # Before this line is run, it does not include PrivateKey (this class),
-        # # and after this line is run, it will include PrivateKey.  This should
-        # # probably be done with some manner of metaclass decorator instead.
-        # #
-        # # Before the next two lines are run, this is the situation:
-        # # > cls.__bases__
-        # #    (<class 'conda_content_trust.common.MixinKey'>,
-        # #     <class 'cryptography.hazmat.primitives.asymmetric.ed25519.Ed25519PrivateKey'>)
-        # # > new_object.__class__
-        # #    <class 'cryptography.hazmat.backends.openssl.ed25519._Ed25519PrivateKey'>
-        # cls.__bases__ += tuple()
-
-        new_object.__class__ = cls
-
-        assert isinstance(new_object, cls)
-        assert isinstance(new_object, ed25519.Ed25519PrivateKey) or isinstance(
-            new_object, ed25519.Ed25519PublicKey
-        )
-
-        checkformat_key(new_object)
-        return new_object
+        if type(k1) is not type(k2):
+            return False
+        return cls.to_bytes(k1) == cls.to_bytes(k2)
```

Comment:
Hm can we simplify this?

```
return type(k1) == type(k2) == cls and cls.to_bytes(k1) == cls.to_bytes(k2)
```

Reply:
Sure, not sure if it's more readable, but it does the same :)